include/aws/common/encoding.h

```diff
@@ -39,7 +39,7 @@ int aws_hex_compute_encoded_len(size_t to_encode_len, size_t *encoded_length);
  * output.
  */
 AWS_COMMON_API
-int aws_hex_encode(const struct aws_byte_buf *AWS_RESTRICT to_encode, struct aws_byte_buf *AWS_RESTRICT output);
```

Comment:
I'll note that we are inconsistent wrt whether inputs or outputs go first between the core byte_buf/byte_cursor APIs and the encoding APIs. While we're repaving the earth, can we fix this as well?

Reply:
Yup!

Reply:
Hmm. In cases where we're doing a conversion, there's no clear dividing line that would say that the input is the thing we're performing the action on, vs the output buffer. Remember that the C standard library uses the convention that output buffers always come first; why should we not follow that here?

Reply:
Did we come to a consensus on this?

Reply:
I'm with Henso on this one, I like having the out param last. Plus, in the case of the encoding functions, it's easier to think of the input as the `this` param than the output. I.E. I think this:
```c
arrayOfThings.encode(outbuf);
```
is clearer than:
```c
outbuf.encode(arrayOfThings);
```